src/issue-activity.ts

```diff
@@ -19,12 +19,42 @@ import {
 } from "./start";
 
 export enum CommentType {
+  /**
+   * Review related item
+   */
   REVIEW = 0b1,
+  /**
+   * Issue related item
+   */
   ISSUE = 0b10,
+  /**
+   * User assigned to the {@link CommentType.ISSUE} or {@link CommentType.REVIEW}
+   */
   ASSIGNEE = 0b100,
+  /**
+   * The author of the {@link CommentType.ISSUE} or {@link CommentType.REVIEW}
+   */
   ISSUER = 0b1000,
+  /**
+   * A user that is part of the organization or owner of the repo
+   */
   COLLABORATOR = 0b10000,
+  /**
+   * A user that is NOT part of the organization nor owner of the repo
+   */
   CONTRIBUTOR = 0b100000,
+  /**
+   * A user comment action on a {@link CommentType.ISSUE} or {@link CommentType.REVIEW}
+   */
+  COMMENTED = 0b1000000,
+  /**
+   * Pull request opening item
+   */
+  TASK = 0b10000000,
```

Comment:
Shouldn't task be issue opening item and specification is pull request opening item? At least that's how I understand from the code.

Reply:
I made it according to the previous version:
- Issue Specification means the item when opening an issue
- Issue Task means the item that fixed the issue

Example with the current comments:
https://github.com/ubiquibot/comment-incentives/issues/31#issuecomment-1998609326

Reply:
I can see from the example that it works correctly but when reading the code I get confused
https://github.com/ubiquibot/conversation-rewards/pull/12/files#diff-03f6b1f98f9f1df2651b65c46241c43e65d9544f7ea1de71e94255e757b3f2f0R108
as I understand the if statement will be true if the comment is the first/opening comment of the issue or pull request so then it checks if it's issue then it will be TASK otherwise (if it's PR) it will be SPECIFICATION?

Reply:
That is correct. I understand now why you're confused:

The value for the task would be
`"ISSUE|ISSUER|TASK"`
because it is inside the `issue`

The value for the Specification would be
`"REVIEW|ISSUER|SPECIFICATION"`
because it is inside the `pull`

I display `Issue: Specification` and `Issue: Task` just to stick to our current display format. Maybe instead I could display `Review: Task` instead?

Reply:
Spoke a bit on this in dms. @whilefoo my intent with referring to GitHub Issues as "Tasks" was to emphasize that its a funded, and recognized "real project" so I want to refer to them as "Tasks" 

Since we are dealing so heavily with the GitHub API, I figured that there is benefit to being specific about what words we are using. Sure we can load in all the GitHub Issues on a repository, but not all of them will be funded and recognized. We are only primarily concerned with working with "Tasks"

Reply:
@whilefoo you are correct, these were wrongly used. I corrected `TASK` and replaced it by `SPECIFICATION` so we have:
`"ISSUE|ISSUER|SPECIFICATION"` and `"REVIEW|ISSUER|TASK"`

I used `ISSUER` for this because technically the issuer is not necessarily the assignee in both cases.

Reply:
this makes more sense!